packages/macros/README.md

```diff
@@ -8,61 +8,179 @@ Traditionally, Ember addons have a lot of power to run arbitrary code during the
+}
+```
+
+### importSync
+
+The primary reason for Embroider's existence is to create statically analyzable builds. An under pinning of this
+is the ability to walk and understand the dependency graph of every module. Embroider can natively understand `imports` such as `import foo as 'foo'` but cannot handle `require`'s (imagine: `require(bar ? 'bar' : 'baz')`. The `importSync` macro is way to "tell" Embroider about the existence of a module and to bring it into a package's scope such that it can be discovered and included into the final build. `importSync` takes a single static string as its only required argument.
+
+```js
+import { importSync } from '@embroider/macros';
+
+let foo = importSync('foo');
+
+// will compile to:
+
+let foo = require('foo');
+```
+
 ### dependencySatisfies
 
-Tests whether a given dependency is present and satisfies the given semver range.
+Tests whether a given dependency is present and satisfies the given semver range. Both arguments must be strings and the second argument will be passed into [semver's satisfies](https://github.com/npm/node-semver#usage) method.
 
-## Setting Configuration: from an Ember app
+```js
+import { dependencySatisfies } from '@embroider/macros';
 
-1. Add `@embroider/macros` as `devDependency`.
-2. In `ember-cli-build.js`, do:
+let doesFooExist = dependencySatisfies('foo', '1.0.0');
 
-   ```js
-   let app = new EmberApp(defaults, {
-     '@embroider/macros': {
-       // this is how you configure your own package
-       setOwnConfig: {
-         // your config goes here
-       },
-       // this is how you can optionally send configuration into your
-       // dependencies, if those dependencies choose to use
-       // @embroider/macros configs.
-       setConfig: {
-         'some-dependency': {
-           // config for some-dependency
-         },
-       },
-     },
-   });
-   ```
+// will compile to:
 
-## Setting Configuration: from an Ember Addon
+let doesFooExist = true; // or false if the dependency was not satisfied
+```
 
-1. Add `@embroider/macros` as `dependency`.
-2. In `index.js`, do:
+We can use this macro along with the `macroCondition` and `importSync` macro's from above to do something more complex:
+
+```js
+import { macroCondition, dependencySatisfies, importSync } from '@embroider/macros';
+
+if (macroCondition(dependencySatisfies('ember-qunit', '*'))) {
+  return importSync('ember-qunit');
+} else if (macroCondition(dependencySatisfies('ember-mocha', '*'))) {
+  return importSync('ember-mocha');
+}
+```
+
+```hbs
+{{macroDependencySatisfies 'qunit' '^2.8.0'}}
+```
+
+### getOwnConfig, getConfig, and getGlobalConfig
+
+A common pattern is to have a set of configuration properties that you define (or a consumer defines for you) which you base certain build time conditions around. This is achieved via the `getOwnConfig`, `getConfig`, and `getGlobalConfig` macros (depending on which config you want to read).
```

Comment:
Should we explain those in a bit more detail? Like `getGlobalConfig` was new to me, and not mentioned in the RFC, so it's not obvious what a global config is.